pkg/bff/mock/admin.go

```diff
@@ -0,0 +1,162 @@
+package mock
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/gin-gonic/gin"
+	"github.com/trisacrypto/directory/pkg/bff/admin"
+	apiv2 "github.com/trisacrypto/directory/pkg/gds/admin/v2"
+	"github.com/trisacrypto/directory/pkg/gds/tokens"
+)
+
+const (
+	StatusEP           = "Status"
+	ListCertificatesEP = "ListCertificates"
+)
+
+func NewAdmin() (a *Admin, err error) {
+	a = &Admin{}
+
+	// Create the mock token manager for authentication
+	if a.tokens, err = tokens.MockTokenManager(); err != nil {
+		return nil, err
+	}
+
+	// Configure the router for the mock
+	gin.SetMode(gin.TestMode)
+	a.router = gin.New()
+	a.authorize = apiv2.Authorization(a.tokens)
+	a.csrf = apiv2.DoubleCookie()
+	a.setupHandlers()
+
+	a.srv = httptest.NewTLSServer(a.router)
+	if a.client, err = admin.NewMock(a.tokens, a.srv); err != nil {
+		return nil, err
+	}
+
+	return a, nil
+}
+
+// Admin is a mock implementation the DirectoryAdministrationServer using an httptest
+// server. The Client() method provides access to the DirectoryAdministrationClient
+// that can be used by tests to make calls to the Admin API using the BFF client. This
+// allows the BFF tests to set specific responses for individual endpoints. To
+// configure the behavior for specific endpoints, use the UseFixture, UseError, or
+// UseHandler methods.
+type Admin struct {
```

Comment:
Nice - I like this approach of creating a universal mock with `httptest` in the same way that we used `bufconn` with our `gRPC` universal mocks. It definitely makes the tests easier to read! 

Could we move the mock code into `pkg/bff/admin/mock/mock.go` to separate the test code from the primary package? One thing that folks seem to be confused by a lot is what is a mock (e.g. a thing that stands in for another thing during tests) and what is mocked (e.g. a thing configured with internal mock objects). I'm trying to think about how to reorganize things better; for example, potentially the mock would go into `pkg/gds/mock` because what we're mocking is the GDS admin, but I'm not sure. Since we may move things in the future, isolating this into its own package would help that process.